fetch/content-type/resources/content-types.json

```diff
@@ -0,0 +1,104 @@
+[
+  {
+    "input": ["", "text/plain"],
+    "encoding": null,
+    "mimeType": "text/plain",
+    "documentContentType": "text/plain"
+  },
+  {
+    "input": ["text/plain", ""],
+    "encoding": null,
+    "mimeType": "text/plain",
+    "documentContentType": "text/plain"
+  },
+  {
+    "input": ["text/html", "text/plain"],
+    "encoding": null,
+    "mimeType": "text/plain",
+    "documentContentType": "text/plain"
+  },
+  {
+    "input": ["text/plain;charset=gbk", "text/html"],
+    "encoding": null,
+    "mimeType": "text/html",
+    "documentContentType": "text/html"
+  },
+  {
+    "input": ["text/plain;charset=gbk", "text/html;charset=windows-1254"],
+    "encoding": "windows-1254",
+    "mimeType": "text/html;charset=windows-1254",
+    "documentContentType": "text/html"
+  },
+  {
+    "input": ["text/plain;charset=gbk", "text/plain"],
+    "encoding": "GBK",
+    "mimeType": "text/plain;charset=gbk",
+    "documentContentType": "text/plain"
+  },
+  {
+    "input": ["text/plain;charset=gbk", "text/plain;charset=windows-1252"],
+    "encoding": "windows-1252",
+    "mimeType": "text/plain;charset=windows-1252",
+    "documentContentType": "text/plain"
+  },
+  {
+    "input": ["text/html;charset=gbk", "text/html;x=\",text/plain"],
+    "encoding": "GBK",
+    "mimeType": "text/html;x=\",text/plain\";charset=gbk",
+    "documentContentType": "text/html"
+  },
+  {
+    "input": ["text/html;charset=gbk", "text/plain", "text/html"],
+    "encoding": null,
+    "mimeType": "text/html",
+    "documentContentType": "text/html"
+  },
+  {
+    "input": ["text/plain", "*/*"],
+    "encoding": null,
+    "mimeType": "text/plain",
+    "documentContentType": "text/plain"
+  },
+  {
+    "input": ["text/html", "*/*"],
+    "encoding": null,
+    "mimeType": "text/html",
+    "documentContentType": "text/html"
+  },
+  {
+    "input": ["*/*", "text/html"],
+    "encoding": null,
+    "mimeType": "text/html",
+    "documentContentType": "text/html"
+  },
+  {
+    "input": ["text/plain", "*/*;charset=gbk"],
+    "encoding": null,
+    "mimeType": "text/plain",
+    "documentContentType": "text/plain"
+  },
+  {
+    "input": ["text/html", "*/*;charset=gbk"],
+    "encoding": null,
+    "mimeType": "text/html",
+    "documentContentType": "text/html"
+  },
+  {
+    "input": ["text/html;x=\"", "text/plain"],
+    "encoding": null,
+    "mimeType": "text/html;x=\", text/plain\"",
+    "documentContentType": "text/html"
+  },
+  {
+    "input": ["text/html;\"", "text/plain"],
+    "encoding": null,
+    "mimeType": "text/html",
+    "documentContentType": "text/html"
+  },
+  {
+    "input": ["text/html;\"", "\\\"", "text/plain"],
+    "encoding": null,
+    "mimeType": "text/html",
+    "documentContentType": "text/html"
+  }
```

Comment:
This seems like a pretty complete list to me.  Some suggestions:

// Make sure values other than charset are discarded from earlier lines (Should we test "boundary", too?  It's the only other value supported in browsers, so could imagine boundary and charset behaving the same, while every other values behaved differently):
"input": ["text/plain;charset=gbk;x=foo", "text/plain"] => "text/plain; charset=GBK"

// Make sure close quotes work across input lines.
"input": ["text/html;\"", "\\\"", "text/plain", "\";charset=GBK"] => "text/html; charset=GBK"

// This is largely the same as above, but for the MIME type itself, and without the extra entire quoted line - don't think it adds a whole lot of value, but still think it's nice to have the close quote and the final value on different lines.  Could also do a variant of this without the "\"", but think the value there is even more minimal than this one.
"input": ["text/html;\"",  "\"", "text/plain"] => "text/plain"

// Make sure that quoted MIME types don't work.  A case testing this perhaps fits better in wpt/mimesniff/mime-types/resources/mime-types.json.
["text/plain", "\"text/html\""] => "text/plain"